docs/source/notes/cuda.rst

```diff
@@ -513,3 +513,189 @@ by GIL of Python interpreter.
+work, and (if ``autograd_aware=True``) also automatically creates graphs of their
+backward-pass work. See
+:ref:`Partial-network capture<partial-network-capture>` for examples.
+
+.. _capture-constraints:
+
+Constraints
+~~~~~~~~~~~
+
+* Capture must occur on a non-default stream. (This is only a concern if you use the raw
+  :meth:`CUDAGraph.capture_begin<torch.cuda.CUDAGraph.capture_begin>` and
+  :meth:`CUDAGraph.capture_end<torch.cuda.CUDAGraph.capture_end>` calls.
+  :class:`~torch.cuda.graph` and
+  :func:`~torch.cuda.make_graphed_callables` set a side stream for you.)
+* Within a process, only one capture may be underway at a time.
+* No non-captured CUDA work may run in this process (on any thread) while capture is underway.
+* CPU work is not captured. If the captured ops include CPU work, that work will be elided during replay.
+* Ops that sychronize the CPU with the GPU (e.g., ``.item()`` calls) are prohibited.
+* Every replay reads from and writes to the same (virtual) memory addresses.
+* Dynamic control flow (based on CPU or GPU data) is prohibited.
+* Dynamic shapes are prohihited. The graph assumes every tensor in the captured op sequence
+  has the same size and layout in every replay.
+* CUDA RNG ops are allowed, but must use default generators. For example, explicitly constructing a
+  new :class:`torch.Generator` instance and passing it as the ``generator`` argument to an RNG function
+  is prohibited.
+* Using multiple streams in a capture is allowed, but there are :ref:`restrictions<multistream-capture>`.
+
+Non-constraints
+~~~~~~~~~~~~~~~
+
+* Once captured, the graph may be replayed on any stream.
+
+.. _basic-capture:
+
+Basic capture and replay
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Example::
+
+    fdsa
+
+.. _multistream-capture:
+
+Using multiple streams
+~~~~~~~~~~~~~~~~~~~~~~
+
+Capture mode automatically propagates to any streams that sync with a capturing stream.
+Within capture, you may expose parallelism by issuing calls to different streams,
+but the overall stream dependency DAG must branch out from the
+initial capturing stream after capture begins and rejoin the initial stream
+before capture ends::
+
+    with torch.cuda.graph(G):
+        # on context manager entrance, torch.cuda.current_stream()
+        # is the initial capturing stream
+
+        # INCORRECT (does not branch out from or rejoin initial stream)
+        with torch.cuda.stream(s):
+            cuda_work()
+
+        # CORRECT:
+        # branches out from initial stream
+        s.wait_stream(torch.cuda.current_stream())
+        with torch.cuda.stream(s):
+            cuda_work()
+        # rejoins initial stream before capture ends
+        torch.cuda.current_stream().wait_stream(s)
+
+.. _full-iteration-capture:
+
+Full-iteration capture
+^^^^^^^^^^^^^^^^^^^^^^
+
+.. _partial-network-capture:
+
+Partial-network capture
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If some of your network is unsafe to capture (e.g., due to dynamic control flow,
```

Comment:
some part of your network

Reply:
I like "some" here because more than 1 "part" of the network may be capture-unsafe.